components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -495,7 +516,16 @@ run (const std::string& filename,
         count.sync_to_host();
 
         auto func_start = std::chrono::steady_clock::now();
-        scorpio::write_var(filename,count.name(),count.get_internal_view_data<int,Host>());
+        if (m_transpose) {
+          const auto& fl = count.get_header().get_identifier().get_layout().to_string();
+          const std::string helper_name   = "transposed_"+fl;
+          auto& temp = m_helper_fields.at(helper_name);
```

Comment:
Related to my comment above, I don't see how this can work correctly. When below you call `temp.get_internal_view_data<int,Host>()`, the code should throw, as `int` is not the data type of `temp`, since you create all helper fields with the "default" data type, which is `RealType`.